src/main/java/ch/njol/skript/variables/VariablesStorage.java

```diff
@@ -169,6 +191,8 @@ protected <T> T getValue(SectionNode sectionNode, String key, Class<T> type) {
 		}
 	}
 
+	private static final ArrayList<File> registeredFiles = new ArrayList<File>();
```

Comment:
Since you're comparing against their uniqueness and not needing the order, a Set can be used here.
```suggestion
	private static final Set<File> registeredFiles = new HashSet<>();
```
May need import